app/src/main/java/com/example/wellfed/ingredient/IngredientStorageFragment.java

```diff
@@ -19,184 +24,240 @@
 import com.example.wellfed.R;
 import com.example.wellfed.common.Launcher;
 import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textview.MaterialTextView;
 
+import java.util.ArrayList;
+import java.util.Objects;
 
-/**
- * The fragment that displays all ingredients in storage.
- */
-public class IngredientStorageFragment extends Fragment
-        implements Launcher, StorageIngredientAdapter.OnItemClickListener {
-    /**
-     * The ingredientController is the controller for the ingredient.
-     */
-    private IngredientStorageController controller;
-    /**
-     * The recycler view for the ingredients.
-     */
-    RecyclerView recyclerView;
-    /**
-     * The text input for the search bar.
-     */
-    int position;
-    /**
-     * The selected item
-     */
-    private int selected;
-
-    /**
-     * ActivityResultLauncher for the IngredientEditActivity to edit an
-     * ingredient.
-     * The result is a StorageIngredient.
-     * The result is null if the user cancels the edit.
-     */
-    ActivityResultLauncher<StorageIngredient> launcher =
-            registerForActivityResult(new IngredientContract(), result -> {
-                if (result == null) {
-                    return;
-                }
-                String type = result.first;
-                StorageIngredient ingredient = result.second;
-                switch (type) {
-                    case "delete":
-                        controller.deleteIngredient(ingredient);
-                        break;
-                    case "edit":
-                        controller.updateIngredient(ingredient);
-                        break;
-                    case "launch":
-                        this.launch(this.selected);
-                        break;
```

Comment:
We still want to relaunch the view activity when edit is closed

Reply:
I have updated it for the StorageIngredients. Will need to fix it for recipes and mealplan as well but that should be done in a different branch.

Reply:
I will fix it, I meant we should launch the activity using the Ingredient

Reply:
I have fixed it to use `StorageIngredient` to launch it